### ISSUE TITLE
MissingSwitchDefault false positive on switch expressions when wrapped in return()

### ISSUE THREAD

**Issue by jacob-pro** [checkstyle/checkstyle#15504]
- [X] I have read check documentation: https://checkstyle.org/checks/xxxxxx/nameofaffectedcheck.html
- [X] I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
- [X] I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
$ javac Test.java && echo "Success"
Success

$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">

    <module name="TreeWalker">
        <module name="MissingSwitchDefault"/>
    </module>

</module>

$ cat Test.java
import java.util.Optional;

class Test {
    enum Case {
        One,
        Two,
    }

    Optional<String> faulty(Case value) {
        return (
            switch (value) {         // <- MissingSwitchDefault (line 11)
                case One -> Optional.of("One");
                case Two -> Optional.of("Two");
            }
        ).map(String::toUpperCase);
    }

    Optional<String> working(Case value) {
        var x = switch (value) {    // <- This one is fine
            case One -> Optional.of("One");
            case Two -> Optional.of("Two");
        };
        return x.map(String::toUpperCase);
    }
}

$ java "-Duser.language=en -Duser.country=US" -jar checkstyle-10.17.0-all.jar -c config.xml Test.java 
Starting audit...
[ERROR] /home/jdhalsey/IdeaProjects/test-http/src/main/java/Test.java:11:13: switch without "default" clause. [MissingSwitchDefault]
Audit done.
Checkstyle ends with 1 errors.
```

---

**Describe what you expect in detail.**

Should not report a MissingSwitchDefault

---

**Comment by romani** [comment]
@mahfouz72 , please look at this

**Comment by mahfouz72** [comment]
I agree with this being an issue.

From check documentation: 

> This check does not validate any switch expressions. Rationale: The compiler requires switch expressions to be exhaustive. This means that all possible inputs must be covered.

https://github.com/checkstyle/checkstyle/blob/66398245db66758bf07844b180e027b2a570a27a/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MissingSwitchDefaultCheck.java#L171-L174

The check fails to detect line 11 as a switch expression because it is wrapped inside parens

```
        |       |--LITERAL_RETURN -> return [16:8]
        |       |   |--EXPR -> EXPR [21:13]
        |       |   |   `--METHOD_CALL -> ( [21:13]
        |       |   |       |--DOT -> . [21:9]
        |       |   |       |   |--LPAREN -> ( [16:14]
        |       |   |       |   |--LITERAL_SWITCH -> switch [17:12]

```

**Comment by mahfouz72** [comment]
@jacob-pro please update the issue title to be more accurate as we use it in release notes. actually, this code is compile able since Java 14.

`MissingSwitchDefault false positive on Java 21 Switch statement when wrapped in return()` => `MissingSwitchDefault false positive on switch expressions when wrapped in return()`

**Comment by rnveach** [comment]
@nrmancuso I am good to approve.
